lock-api/src/main/java/com/palantir/lock/client/NamespacedCoalescingLeaderTimeGetter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.lock.v2.LeaderTime;
+
+public class NamespacedCoalescingLeaderTimeGetter implements LeaderTimeGetter {
```

Comment:
I'd add some javadocs here, explaining more precisely how this works, as the difference between this and the "obvious" implementation is important and subtle.